### PR TITLE
Small refactor to move temp array to where its used from the Application

### PR DESCRIPTION
**Pull request by mvaligursky**
the list is only used in one function of the entity, so moved it there. It’s not used in the Editor either.

### CODE REVIEW

File: src/framework/entity.js [hunk 1]
```diff
@@ -27,6 +27,9 @@ import { Application } from './application.js';
 /** @typedef {import('./components/sound/component.js').SoundComponent} SoundComponent */
 /** @typedef {import('./components/sprite/component.js').SpriteComponent} SpriteComponent */
 
+/** @type {GraphNode[]} */
```
Comment (willeastcott) [inline src/framework/entity.js:30]: You'll need an @ignore here or this symbol will actually appear in the API reference (in the `pc.` namespace).

File: src/framework/entity.js [hunk 2]
```diff
@@ -465,11 +468,11 @@ class Entity extends GraphNode {
 
         if (enableFirst) {
             // do not cache the length here, as enableList may be added to during loop
-            for (let i = 0; i < this._app._enableList.length; i++) {
-                this._app._enableList[i]._onHierarchyStatePostChanged();
+            for (let i = 0; i < _enableList.length; i++) {
+                _enableList[i]._onHierarchyStatePostChanged();
```
Comment (willeastcott) [inline src/framework/entity.js:472]: Presumably this won't work in the multi-application case?

Comment (mvaligursky) [reply src/framework/entity.js:472]: I thought about it, and I think it should? This is only used inside a single function, those apps don't run in parallel, but in serial.

File: src/framework/entity.js (hunk 1 above)
Comment (mvaligursky) [reply src/framework/entity.js]: done. I assumed those are exposed in docs only if there's a description, not only a type.

**Review by willeastcott — APPROVED** [review]

**Comment by willeastcott** [comment]
I love little refactors like this. Nice job.

**Pull request merged**